modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/ScriptMessageContext.java

```diff
@@ -97,737 +38,71 @@ public ScriptMessageContext(MessageContext mc, XMLHelper xmlHelper) {
-    }
+    Object getPayloadXML() throws ScriptException;
 
     /**
-     * Set the SOAP body payload from XML
+     * Set the SOAP body payload from XML.
      *
      * @param payload Message payload
      * @throws ScriptException For errors in converting xml To OM
      * @throws OMException     For errors in OM manipulation
      */
+    void setPayloadXML(Object payload) throws OMException, ScriptException;
 
-    public void setPayloadXML(Object payload) throws OMException, ScriptException {
-        SOAPBody body = mc.getEnvelope().getBody();
-        OMElement firstChild = body.getFirstElement();
-        OMElement omElement = xmlHelper.toOMElement(payload);
-        if (firstChild == null) {
-            body.addChild(omElement);
-        } else {
-            firstChild.insertSiblingAfter(omElement);
-            firstChild.detach();
-        }
-    }
 
     /**
      * Get the JSON object representation of the JSON message body of the request.
      *
      * @return JSON object of the message body
      */
-    public Object getPayloadJSON() {
-        return jsonObject(mc);
-    }
+    Object getPayloadJSON();
 
-    public Object getJsonText() {
-        if (mc == null) {
-            return "";
-        }
-        Object text = mc.getProperty(JSON_TEXT);
-        return text == null ? "{}" : text;
-    }
 
     /**
-     * Get the Message Payload as a text
+     * Get the Message Payload as a text.
      *
      * @return Payload as text
      */
-    public String getPayloadText() {
-        if (JsonUtil.hasAJsonPayload(((Axis2MessageContext) mc).getAxis2MessageContext())) {
-            return JsonUtil.jsonPayloadToString(((Axis2MessageContext) mc).getAxis2MessageContext());
-        } else {
-            return mc.getEnvelope().toString();
-        }
-    }
+    Object getJsonText();
 
-    /**
-
-    /**
-     * Saves the payload of this message context as a JSON payload.
-     *
-     * @param jsonPayload Javascript native object to be set as the message body
-     * @throws ScriptException in case of creating a JSON object out of
-     *                         the javascript native object.
-     */
-    public void setPayloadJSON0(Object jsonPayload) throws ScriptException {
-        org.apache.axis2.context.MessageContext messageContext;
-        messageContext = ((Axis2MessageContext) mc).getAxis2MessageContext();
-
-        String jsonString;
-        if (jsonPayload instanceof String) {
-            jsonString = (String) jsonPayload;
-        } else {
-            jsonString = serializeJSON(jsonPayload);
-        }
-        try {
-            JsonUtil.getNewJsonPayload(messageContext, jsonString, true, true);
-        } catch (AxisFault axisFault) {
-            throw new ScriptException(axisFault);
-        }
-        //JsonUtil.setContentType(messageContext);
-        Object jsonObject = scriptEngine.eval('(' + jsonString + ')');
-        setJsonObject(mc, jsonObject);
-    }
+    String getPayloadText();
```

Comment:
We should have doc comments for all the methods in a public interface. Shall we fix in all places.

Reply:
Hi Asanka,
I have added the doc comments